gqcpy/src/QCMethod/HF/GHF/GHFSCFEnvironment_bindings.cpp

```diff
@@ -138,13 +111,30 @@ void bindGHFSCFEnvironments(py::module& module) {
     // Provide bindings for real-valued GHF SCF environments.
     py::class_<GHFSCFEnvironment<double>> py_GHFSCFEnvironment_d {module, "GHFSCFEnvironment_d", "An algorithm environment that can be used with real-valued GHF SCF solvers."};
 
-    bindQCModelGHFSCFEnvironmentInterface(py_GHFSCFEnvironment_d);
+    bindQCMethodGHFSCFEnvironmentInterface(py_GHFSCFEnvironment_d);
+    bindQCMethodHartreeFockSCFEnvironmentInterface(py_GHFSCFEnvironment_d);
 
 
     // Provide bindings for complex-valued GHF SCF environments.
     py::class_<GHFSCFEnvironment<complex>> py_GHFSCFEnvironment_cd {module, "GHFSCFEnvironment_cd", "An algorithm environment that can be used with complex-valued GHF SCF solvers."};
 
-    bindQCModelGHFSCFEnvironmentInterface(py_GHFSCFEnvironment_cd);
+    py_GHFSCFEnvironment_cd
+        .def_static(
+            "WithCoreGuessMadeComplex",
+            [](const size_t N, const GSQHamiltonian<complex>& hamiltonian, const ScalarGSQOneElectronOperator<complex>& S) {
+                return GHFSCFEnvironment<complex>::WithCoreGuessMadeComplex(N, hamiltonian, S);
+            },
+            "Initialize an GHF SCF environment with an initial coefficient matrix that is obtained by diagonalizing the core Hamiltonian matrix and subsequently adding/subtracting a small complex value from certain elements.")
+
+        .def_static(
+            "WithCoreGuessMadeComplex",
+            [](const size_t N, const GSQHamiltonian<complex>& hamiltonian, const ScalarGSQOneElectronOperator<complex>& S, const std::function<Eigen::Matrix<complex, Eigen::Dynamic, Eigen::Dynamic>(const Eigen::Matrix<complex, Eigen::Dynamic, Eigen::Dynamic>&)>& transformation_function) {
+                return GHFSCFEnvironment<complex>::WithCoreGuessMadeComplex(N, hamiltonian, S, transformation_function);
+            },
+            "Initialize an GHF SCF environment with an initial coefficient matrix that is obtained by diagonalizing the core Hamiltonian matrix and subsequently using a given transformation function to transform it into a complex valued matrix.");
```

Comment:
I might be wrong, but at first sight it seems that these two bindings can also be re-used in the R and U bindings.

Reply:
The reason I didn't do this initially was, as I recall, because of the difference in arguments between R/GHF and UHF, as UHF takes an `N_a` and `N_b` separately. However I might be able to fix this as the `SpinResolved<size_t>` type has been bound, which would result in the ability to add the environment bindings to the interface I believe. 

Although having to 'make' a `spinResolved<size_t>` in order to pass the alpha and beta electrons to UHF might be somewhat cumbersome when using the python bindings. What do you think?

Reply:
I completely agree. The number of different arguments for these named constructors was an oversight on my part!